### ISSUE TITLE
Issue in mAP calculation

### ISSUE THREAD

**Issue by ktertikas** [LongLong-Jing/Cross-Modal-Center-Loss#11]
Hello, thank you for your code!

I am having a look at the way that the mAP is being calculated, in the following function:
https://github.com/LongLong-Jing/Cross-Modal-Center-Loss/blob/5025837561a5e059940462edde99aafe80680871/evaluate_retrieval.py#L94

It seems to me that the calculation has a small issue, when you do the calculation for `Image2Image`, `Mesh2Mesh`, and `Point2Point`. In particular, I believe that you should remove the query item from the distance calculations, because otherwise you would always have the diagonal distance matrix value being 0 (it will be the same item in both query and database), so this would produce a slightly wrong mAP value.

Thanks,
Konstantinos

**Comment by piseabhijeet** [comment]
@ktertikas 

You are right and i had noticed that too. The code should be changed for the same modalities calculation.
